src/Symfony/Component/Ldap/Adapter/ExtLdap/EntryManager.php

```diff
@@ -64,4 +64,23 @@ public function remove(Entry $entry)
             throw new LdapException(sprintf('Could not remove entry "%s": %s', $entry->getDn(), ldap_error($con)));
         }
     }
+
+    /**
+     * Renames an entry on the Ldap server.
+     *
+     * @param Entry  $entry
+     * @param string $newRdn
+     * @param string $newParent
+     * @param bool   $shouldDeleteOldRdn
+     *
+     * @throws LdapException
+     */
+    public function rename(Entry $entry, $newRdn, $newParent, $shouldDeleteOldRdn = true)
```

Comment:
An `Entry` object should not be needed in order to rename an entry. Also, `$newParent` should be null by default.

More generally speaking, I'm not really fond of simply exposing the same API as the `ldap` PHP extension. This would actually go against the goal of this component, which aims to provide a layer of abstraction on top of multiple implementations of Ldap (the PHP Ldap extension being the first available adapter).

Instead, I would improve the logic in the `update()` method, and add more logic in the `Entry` class in order to be able to compute a changeset, or simply flag the Entry instance as "dirty".

Currently, the logic for the `update()` method is pretty poor, and does not check whether the attributes have changed or not, or if the DN has changed or not. This could be pretty much improved, for Symfony 3.3.